internal/querycoord/querynode.go

```diff
@@ -564,10 +586,10 @@ func (qn *queryNode) loadSegments(ctx context.Context, in *querypb.LoadSegmentsR
 }
 
 func (qn *queryNode) releaseSegments(ctx context.Context, in *querypb.ReleaseSegmentsRequest) error {
-	qn.serviceLock.RLock()
-	onService := qn.onService
-	qn.serviceLock.RUnlock()
-	if !onService {
+	qn.stateLock.RLock()
+	state := qn.state
+	qn.stateLock.RUnlock()
+	if state != online {
```

Comment:
why not add a function named getState() or isOnline()?

Reply:
Isn't it the same？

Reply:
Simplify the code